collector/collector_linux_test.go

```diff
@@ -0,0 +1,41 @@
+package collector
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/coreos/go-systemd/journal"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSystemd_Collect(t *testing.T) {
+	if os.Getenv("TRAVIS") != "" {
+		t.Skipf("SKIPPING: We can not read from journal in Travis")
+	}
+	if !journal.Enabled() {
+		t.Skipf("SKIPPING: Journal not enabled")
+	}
+
+	err := journal.Send("test message", journal.PriInfo, map[string]string{"UNIT": "test-unit"})
+	require.NoError(t, err)
+
+	c := NewSystemd(
+		"test",
+		false,
+		"test-unit",
+		time.Second,
+	)
+	r, err := c.Collect(context.TODO())
```

Comment:
This might introduce a race condition if the message is not picked up fast enough by journald, right?

Reply:
Right but I think it's more predictable then relying on `systemd-journal.service` to log something